src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
+      victim.health -= 50;
+    }
+
+    const deadAnimal = Animal.alive.indexOf(victim);
+
+    if (victim.health === 0) {
+      Animal.alive.splice(deadAnimal, 1)
+    }
```

Comment:
Try to avoid nesting loops (indexOf + splice), use filter instead - checklist.

Reply:
2 mentors - 2 options))  I wrote it with filter at first, but AndriiRut asked me to change it )))))